Frameworks/Foundation/NSKeyValueCoding.mm

```diff
@@ -477,6 +486,11 @@ - (void)setValue:(id)val forKey:(NSString*)key {
     }
 
     const char* rawKey = [key UTF8String];
+
+    if (!val) {
+        switch (rawKey[0]) { OBJC_APPLY_NUMERIC_TYPE_ENCODINGS(FAIL_IF_SCALAR); }
```

Comment:
I don't get it. Can't you just call `[self setNilValueForKey:]`? Or is this to get the appropriate "is not key-value coding compliant" exception?